src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs

```diff
@@ -46,12 +46,22 @@ internal static class RemoteHostClientExtensions
             this RemoteHostClient client, string serviceName, Solution solution, object callbackTarget, CancellationToken cancellationToken)
         {
             var connection = await client.TryCreateConnectionAsync(serviceName, callbackTarget, cancellationToken).ConfigureAwait(false);
-            if (connection == null)
+            try
             {
-                return null;
-            }
+                if (connection == null)
+                {
+                    return null;
+                }
 
-            return await SessionWithSolution.CreateAsync(connection, solution, cancellationToken).ConfigureAwait(false);
+                using (var boxed = connection.Box())
```

Comment:
can you explain when people use box? is it like we inside of me take ownership from me then noop otherwise I will dispose it kind of thing?

Reply:
➡️ Ownership of an `IDisposable` resource will exist at one location in memory with this pattern. Normally this location is stored on the evaluation stack for efficiency (within the struct). However, it is not always possible to reference a memory location on the evaluation stack (e.g. when calling an `async` method), so a boxed location is provided as a fallback to allow explicit ownership transfer in these cases.